cachito/workers/pkg_managers/rubygems.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import re
+from dataclasses import dataclass
+from pathlib import Path
+
+from gemlock_parser.gemfile_lock import GemfileLockParser
+
+from cachito.errors import ValidationError
+
+GIT_REF_FORMAT = re.compile(r"^[a-fA-F0-9]{40}$")
+PLATFORMS_RUBY = re.compile(r"^PLATFORMS\n {2}ruby\n\n", re.MULTILINE)
+
+
+@dataclass
+class GemMetadata:
+    """Gem metadata."""
+
+    name: str
+    version: str
+    type: str
+    path: str
+
+
+def parse_gemlock(source_dir, gemlock_path):
+    """Parse dependencies from Gemfile.lock.
+
+    :param Path source_dir: the full path to the project directory
+    :param Path gemlock_path: the full path to Gemfile.lock
+    :return: list of Gems
+    """
+    if not gemlock_path.is_file():
+        return []
```

Comment:
Should missing lockfile be an error? Or is it a valid case for projects that use bundler?

Reply:
It is required to have `Gemfile`, and therefore we could say that the project is required to have Gemfile.lock.

I noticed that implementation for Python doesn't throw any error too, so I wanted to keep it similar, but yeah, raising an exception makes sense.